api/src/main/java/edu/cornell/mannlib/vitro/webapp/search/controller/AutocompleteController.java

```diff
@@ -99,6 +101,8 @@ else if ( parts.length == 1 ) {
                 return;
             }
             log.debug("query for '" + qtxt +"' is " + query.toString());
+            Set<String> currentRoles = SearchFiltering.getCurrentUserRoles(vreq);
```

Comment:
Again I am thinking here whether it is possible to get currentRoles once, not as reaction on any single Ajax event. I mean when a user type a couple of letters this will be called, and then when add one additional letter, getting roles and applying filters will be invoked again

Reply:
I think we should get it at least once per request.